src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/tests/test_batch_mgmt_commands.py

```diff
@@ -78,6 +79,18 @@ def body(self):
 
         self.assertTrue(keys['primary'] != keys2['primary'])
 
+        self.cmd('batch account login -g {} -n {}'.
+                 format(rg, name), checks=NoneCheck())
+        self.assertEqual(az_config.config_parser.get('batch', 'auth_mode'), 'aad')
+        self.assertEqual(az_config.config_parser.get('batch', 'account'), name)
+        self.assertFalse(az_config.config_parser.has_option('batch', 'access_key'))
+
+        self.cmd('batch account login -g {} -n {} --shared-key-auth'.
+                 format(rg, name), checks=NoneCheck())
+        self.assertEqual(az_config.config_parser.get('batch', 'auth_mode'), 'shared_key')
+        self.assertEqual(az_config.config_parser.get('batch', 'account'), name)
+        self.assertEqual(az_config.config_parser.get('batch', 'access_key'), keys2['primary'])
```

Comment:
Should mock az_config.config_parser in tests.

As is, if I run this test on my machine, I believe it will change my `~/.azure/config` file which is an unintended consequence as `az batch account login` will set some config values.

You should be able to mock az_config.config_parser or if not, maybe you can mock the config file used for the test as in `test_config.py` to use a tmp file.